receiver/jaegerreceiver/trace_receiver.go

```diff
@@ -300,14 +322,30 @@ func (jr *jReceiver) GetBaggageRestrictions(serviceName string) ([]*baggage.Bagg
 	return nil, nil
 }
 
-func (jr *jReceiver) startAgent() error {
+func (jr *jReceiver) PostSpans(ctx context.Context, r *api_v2.PostSpansRequest) (*api_v2.PostSpansResponse, error) {
+	ctxWithReceiverName := observability.ContextWithReceiverName(ctx, collectorReceiverTagValue)
+
+	td, err := jaegertranslator.ProtoBatchToOCProto(r.Batch)
+	td.SourceFormat = "jaeger"
+	if err != nil {
+		observability.RecordTraceReceiverMetrics(ctxWithReceiverName, len(r.Batch.Spans), len(r.Batch.Spans))
+		return nil, err
+	}
+
+	err = jr.nextConsumer.ConsumeTraceData(ctx, td)
+	observability.RecordTraceReceiverMetrics(ctxWithReceiverName, len(r.Batch.Spans), len(r.Batch.Spans)-len(td.Spans))
+
+	return &api_v2.PostSpansResponse{}, err
```

Comment:
Great catch! I just checked the reference code in the Jaeger repo and there's indeed an error check there, returning nil in case of error. I just changed this PR to match the reference behavior.